### DISCUSSION TITLE
Provide custom save states for specific challenges

### DISCUSSION THREAD

**Discussion by dude1286** [RetroAchievements/RAIntegration#782] [Ideas]
A lot of lists today have achievements for specific areas of a game that require the player to replay the game from the beginning to get back to. For some games, this doesn't take a long time, while others could take days to get back to again. One example could be no-hit boss achievements in the Zelda games. It can take a long time to get back to these bosses to reattempt the no-hit achievement if they save past the dungeon completion. An idea I have is to create a Save State for a specified achievement, maybe even some leader boards, in the same way that Rare Replay does the Snapshot system. It is a save state that does a specific level and ends either by a timer, a goal has been accomplished, or a failure flag happens. We could also tie-in that a player cannot access said Snapshot until another achievement has been unlocked. Going back to the Zelda example, the player would have to be have the dungeon completion achievement before they can tackle the boss snapshot. This would also give the developer control over what a player will have in their inventory when they reattempt a boss this way, which could make it harder. In ALttP, the cape is a great item to make later bosses easier to no-hit complete, but it is an optional item. The developer could make all the snapshots not have it in the player's inventory, thus incentivizing player's to complete the achievement in their own full play through of the game. This could also open up more leader board ideas as well, like how fast a player can beat a boss without getting hit. It would also make reattempts easier for Hardcore players as just using the Snapshot will not disable Hardcore mode.

**Comment by Jamiras** [comment]
Something to consider when thinking about this: save states are not compatible between emulators, so we'd need a separate one for ralibretro vs. rasnes9x. Additionally, they're rarely compatible between machines (i.e. RetroArch-Windows vs. RetroArch-Android vs. RetroArch-iOS, etc). And sometimes, they aren't compatible between versions (a new version of an emulator or core may not be able to read old save states).
